### PR TITLE
WIP price-fetch from oracle

### PR DESCRIPTION
**Pull request by martinfridrich**
!!!Draft(mvp) price fetch pallet from diadata

### CODE REVIEW

File: pallets/price-fetch/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+[package]
+authors = ['GalacticCoucil']
```
Comment (apopiak) [inline pallets/price-fetch/Cargo.toml:2]:
```suggestion
authors = ['GalacticCouncil']
```

File: pallets/price-fetch/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+				Err(http::Error::Unknown)
+			}
+		}?;
+
+		Ok(price)
+	}
+
+	/// Parse json response body received from dia request
+	///
+	/// Returns `None` when parsing failed or `Some(DiaPriceRecord)` when parsing is successful.
```
Comment (apopiak) [inline pallets/price-fetch/src/lib.rs:387]: na, it panics xD

File: pallets/price-fetch/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+			let now = <pallet_timestamp::Module<T>>::get();
+			Self::deposit_event(RawEvent::NewFetcher(who, SYM.to_vec(), now));
+			Ok(())
+		}
+
+		#[weight = 0]
+		pub fn submit_new_price(origin, price_record: DiaPriceRecord) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(!<Fetchers<T>>::contains_key(&price_record.symbol), Error::<T>::FetcherNotFound);
```
Comment (apopiak) [inline pallets/price-fetch/src/lib.rs:221]:
```suggestion
			ensure!(<Fetchers<T>>::contains_key(&price_record.symbol), Error::<T>::FetcherNotFound);
```

File: pallets/price-fetch/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+			Self::add_new_price_to_list(new_price);
+
+			let now = <pallet_timestamp::Module<T>>::get();
+			Self::deposit_event(RawEvent::NewPricePoint(who, price_record.symbol, now, price_record.price));
+
+			Ok(())
+		}
+
+		#[weight = 0]
+		pub fn submit_new_avg_price(origin, symbol: Symbol, avg_price:Price) -> DispatchResult {
```
Comment (apopiak) [inline pallets/price-fetch/src/lib.rs:239]: this average is not verified, is that on purpose?

File: Cargo.lock [hunk 5]
```diff
@@ -3837,6 +3867,26 @@ dependencies = [
...
+version = "2.0.0"
+dependencies = [
+ "alt_serde",
+ "fixed",
+ "frame-support",
+ "frame-system",
+ "pallet-timestamp",
+ "parity-scale-codec",
+ "primitives",
+ "serde_json 1.0.44",
```
Comment (jak-pan) [inline Cargo.lock:3881]: Looks like we've got some version mismatch here.

File: pallets/price-fetch/src/lib.rs (hunk 2 above)
Comment (martinfridrich) [reply pallets/price-fetch/src/lib.rs:387]: yes, you're right. This comment is how function should work at the end :smile:  I'll fix it

File: pallets/price-fetch/src/lib.rs (hunk 4 above)
Comment (martinfridrich) [reply pallets/price-fetch/src/lib.rs:239]: Verification/validation of submitted values is in progress

File: pallets/price-fetch/src/lib.rs (hunk 2 above)
Comment (martinfridrich) [reply pallets/price-fetch/src/lib.rs:387]: fixed https://github.com/galacticcouncil/hack.HydraDX-node/blob/83b6e02051cc2f3d19b1b351f92174b7d4f83938/pallets/price-fetch/src/lib.rs#L388

**Pull request closed without merging**